apps/barista-design-system/src/shared/services/scroll-spy.service.ts

```diff
@@ -14,212 +14,127 @@
-  private _spiedElementGroups: BaScrollSpiedElementGroup[] = [];
   private _onStopListening = new Subject<void>();
-  private _lastContentHeight: number;
-  private _lastMaxScrollTop: number;
-  private _topOffset = 66;
 
-  // tslint:disable-next-line: no-any
   constructor(
     private _platform: Platform,
     @Inject(DOCUMENT) private _document: any,
   ) {}
 
-  /*
-   * Start tracking a group of elements and emitting active elements; i.e. elements that are
-   * currently visible in the viewport. If there was no other group being spied, start listening for
-   * `resize` and `scroll` events.
-   */
-  spyOn(elements: Element[]): BaScrollSpyInfo | null {
-    if (this._platform.isBrowser && !this._spiedElementGroups.length) {
-      fromEvent(window, 'resize')
-        .pipe(auditTime(300), takeUntil(this._onStopListening))
-        .subscribe(() => {
-          this._onResize();
-        });
-      fromEvent(window, 'scroll')
-        .pipe(auditTime(10), takeUntil(this._onStopListening))
-        .subscribe(() => {
-          this._onScroll();
-        });
-      this._onResize();
-
-      const scrollTop = this._getScrollTop();
-      const maxScrollTop = this._lastMaxScrollTop;
-
-      const spiedGroup = new BaScrollSpiedElementGroup(elements);
-      spiedGroup.calibrate(scrollTop, this._topOffset);
-      spiedGroup.onScroll(scrollTop, maxScrollTop);
-
-      this._spiedElementGroups.push(spiedGroup);
-
-      return {
-        active: spiedGroup.activeScrollItem
-          .asObservable()
-          .pipe(distinctUntilChanged()),
-        unspy: () => {
-          this._unspy(spiedGroup);
-        },
-      };
-    }
-
-    return null;
-  }
-
-  private _getContentHeight(): number {
-    return this._document.body.scrollHeight || Number.MAX_SAFE_INTEGER;
+  /** Start spying on an element array of headlines returning a stream with the active item */
+  spyOn(elements: Element[]): void {
+    // Resize and Scroll trigger event calculates top values and active item.
+    merge(
+      fromEvent(window, 'resize').pipe(
+        auditTime(300),
+        takeUntil(this._onStopListening),
+      ),
+      fromEvent(window, 'scroll').pipe(
+        auditTime(10),
+        takeUntil(this._onStopListening),
+      ),
+    ).subscribe(() => {
+      this._calculateActiveItem(elements);
+    });
   }
 
-  private _getScrollTop(): number {
-    return (this._platform.isBrowser && window.pageYOffset) || 0;
+  /** Calculates top values and finds the active item putting it into a stream */
+  private _calculateActiveItem(elements: Element[]): void {
+    this._calcTopValues(elements);
+    const activeItemId = this.findActiveItemId();
+    if (activeItemId) {
+      this.activeItemId$.next(activeItemId);
+    }
   }
 
-  private _getViewportHeight(): number {
-    return (this._platform.isBrowser && window.innerHeight) || 0;
+  /** Calculates top bounding properties for an element array */
+  private _calcTopValues(elements: Element[]): void {
```

Comment:
NIT: Coding guidelines state to avoid abbreviations.
```suggestion
  private _calculateTopValues(elements: Element[]): void {
```